### PR TITLE
PFB reader update and other changes

### PR DESCRIPTION
**Pull request by ramain**
Long overdue update -  pyfftw in the pfb reader, generalized reader to allow for PFB inversion on old raw data, and allowing for upchannelization of ARO data with incoherent de-dispersion without PFB inversion (this specific case wasn't included in our if statements).

### CODE REVIEW

File: scintellometry/folding/fold.py [hunk 1]
```diff
@@ -238,6 +238,10 @@ def fold(fh, comm, samplerate, fedge, fedge_at_top, nchan,
         else:              # raw.shape=(ntint, nchan*npol)
             raw = raw.reshape(-1, fh.nchan, npol)
 
+        if dedisperse in ['incoherent'] and oversample > 1:
```
Comment (mhvk) [inline scintellometry/folding/fold.py]: Could just be `dedisperse == 'incoherent'`

File: scintellometry/io/arochime.py [hunk 2]
```diff
@@ -285,7 +297,7 @@ def __init__(self, raw_files, blocksize, samplerate, fedge, fedge_at_top,
         (ie: read as dtype (2048,)4bit: 1024)
         """
 
-        self.fh_raw = AROCHIMERawData(raw_files, blocksize, samplerate,
+        self.fh_raw = self._raw_data_class(raw_files, blocksize, samplerate,
                                       fedge, fedge_at_top, time_offset,
```
Comment (mhvk) [inline scintellometry/io/arochime.py]: Also adjust indentation.

File: scintellometry/io/arochime.py [hunk 3]
```diff
@@ -327,39 +338,34 @@ def seek_record_read(self, offset, size):
             raise ValueError("size and offset must be an integer number of records")
 
         raw = self.fh_raw.seek_record_read(offset, size)
-
-        if self.npol == 2:
```
Comment (mhvk) [inline scintellometry/io/arochime.py]: Am a bit confused that this can be removed. I assume it still works on 2-pol data of both the old and new format?

**Comment by ramain** [comment]
@mhvk Made the suggested changes, as well as others I missed in the first commit.

**Comment by mhvk** [comment]
This does now look good, except could you remove that last merge commit: should be
```
git fetch upstream   # just in case
git rebase -i upstream/master
# edit and remove last merge commit, if it is still there
git push --force
```

**Comment by mhvk** [comment]
p.s. How I wish we had test cases...

**Comment by ramain** [comment]
@mhvk I went through the rebase to remove the last merge.  And I agree, test cases are desperately needed for our reduction code...

**Comment by mhvk** [comment]
OK, thanks. I'll merge now...

**Pull request merged**